Sources/PackageLoading/Diagnostics.swift

```diff
@@ -144,9 +141,17 @@ extension Diagnostic.Message {
     }
 }
 
-public struct ManifestLoadingDiagnostic: DiagnosticData {
-    public let output: String
-    public let diagnosticFile: AbsolutePath?
-
-    public var description: String { output }
+extension DiagnosticsMetadata {
+    public var manifestLoadingDiagnosticFile: AbsolutePath? {
```

Comment:
I realize this is temporary conversion, but this "diagnostics file" that then gets sent up to the client is one in particular that I've been eager to convert to use the `.dia` parsing support from TSC to create real diagnostics.

Reply:
that would be great! here I was only trying to model the existing information that was carried in the old API as-is

Reply:
Absolutely, understood.  Just wanted to state what I see as the future direction, to see if you agreed or disagreed.